### DISCUSSION TITLE
Removing a ToolWindow instance

### DISCUSSION THREAD

**Discussion by ItzzLincler** [VsixCommunity/Community.VisualStudio.Toolkit#127] [Q&A] [answered]
Is there a way to remove or delete an instance of a `ToolWindow` ?
I have a `toolwindow` which shows data for a specific solution and I would like it to be destructed after the solution is closed.

**Comment by madskristensen** [comment, marked as answer]
I don't know if that's possible. Why can't you listen for a SolutionClosed event and then redraw your XAML control inside the tool window?

**Reply by ItzzLincler** [reply]
Yeah, that works.
It just requires some extra logic and listening to more events, so I thought there might be a simpler way if you could just Initialize it when you need it.
Then again on second thought it might be bad to initialize it all the time since it requires more overhead.

Thanks anyway :)
